clang/test/Driver/sycl-offload.cpp

```diff
@@ -47,3 +47,60 @@
 // CHECK_S_LLVM: clang{{.*}} "-fsycl-is-device"{{.*}} "-emit-llvm"{{.*}} "-o" "[[DEVICE:.+\.ll]]"
 // CHECK_S_LLVM: clang{{.*}} "-fsycl-is-host"{{.*}} "-emit-llvm"{{.*}} "-o" "[[HOST:.+\.ll]]"
 // CHECK_S_LLVM: clang-offload-bundler{{.*}} "-type=ll"{{.*}} "-inputs=[[DEVICE]],[[HOST]]"
+
+/// Check for default device triple compilations based on object, archive or
+/// forced from command line.
+// RUN:  echo "void foo();" > %t_dummy.cpp
+// RUN:  %clang -fsycl -c %t_dummy.cpp -o %t_dummy.o
+// RUN:  llvm-ar cr %t_dummy.a %t_dummy.o
+// RUN:  touch %t_empty.o
```

Comment:
I believe it's important to check correct behavior for the input archives that contain both an AOT-compiled image (FPGA, primarily) and a SPIR-V image when only the AOT target triple is specified in `-fsycl-targets` on the link command line. Do I understand correctly that the SPIR-V image must also be linked in for such a scenario (unless `-fno-sycl-link-spirv` is passed)?

Reply:
Yup, that is the correct understanding.  That should be what this test is covering, by virtue of checking that the expected targets are being unbundled.

Reply:
Ah, got it now - these checks cover both the SPIR-V-containing archives and cases like FPGA archives with an `.aocx` in the bundle, right? With that, the actual contents of the dummy archive, i.e. whether an actual `.aocx` is present in the bundle, wouldn't matter in terms of these checks.

Reply:
That's right, the aocx and SPIR-V checks are separate.  Of course FPGA aocx checks are only done with the spir64_fpga triple when encountered.